drivers/counter/counter_nxp_mrt.c

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/counter.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/device.h>
+#include <zephyr/irq.h>
+
+#define LOG_MODULE_NAME counter_mrt
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_COUNTER_LOG_LEVEL);
+
+/* Specific for each channel */
+struct nxp_mrt_channel_data {
+	uint32_t top;
+	counter_top_callback_t cb;
+	void *user_data;
+};
+
+/* Shared between all channels */
+struct nxp_mrt_config {
+	struct counter_config_info info;
+	MRT_Type *base;
+	const struct device *clock_dev;
+	clock_control_subsys_t clock_subsys;
+	void (*irq_config_func)(const struct device *dev);
+	struct nxp_mrt_channel_data *data;
```

Comment:
Can you please clarify why the `nxp_mrt_channel_data` is stored inside the config structure and not provided to `dev` structure by passing it to `DEVICE_DT_INST_DEFINE`

Reply:
I'm not sure what you mean. It's not stored inside the config structure, it's stored in RAM, and it is provided to the dev structure (line 247)

Reply:
I am referring to `void *data;` in `struct device`